src/parse.rs

```diff
@@ -98,14 +98,15 @@ pub async fn read_addon_directory<P: AsRef<Path>>(
     root_dir: P,
     flavor: Flavor,
 ) -> Result<Vec<Addon>> {
+    let root_dir = root_dir.as_ref();
+
     // If the path does not exists or does not point on a directory we return an Error.
-    if !root_dir.as_ref().is_dir() {
+    if !root_dir.is_dir() {
         return Err(ClientError::Custom(format!(
             "Addon directory not found: {:?}",
-            root_dir.as_ref().to_owned()
+            root_dir
         )));
     }
-    let root_dir = root_dir.as_ref().to_owned();
```

Comment:
We don't need to make an allocation here, `&Path` worked everywhere it was needed.